### ISSUE TITLE
Embedded Nim HTML (*.enim.html?)

### ISSUE THREAD

**Issue by Willyboar** [nim-lang/needed-libraries#110]
Something similar to Ruby's standard library(erb) or Crystal(ecr).

**Comment by Araq** [comment]
We have htmlgen in the stdlib, "source code filters" in the Nim language and "karax" as a Nimble package. One of these should really fit your needs.

**Comment by Willyboar** [comment]
@Araq yes I agree. Both htmlgen and SCF can make the job ( I haven't look  Karax yet). But I believe embedded nim would be better for new programmers with web programming in mind.

**Comment by disruptek** [comment]
@Araq There's apparently no way to add source code filters to Nim unless you want to maintain your own fork of the compiler, so I'd say that's hardly a reasonable suggestion.

**Comment by enthus1ast** [comment]
Also Nimja https://github.com/enthus1ast/nimja
